### PR TITLE
DEV-14921 - Remove wildcards from IAM policies where possible

### PR DESCRIPTION
**Pull request by along528**

### CODE REVIEW

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

File: main.tf [hunk 1]
```diff
@@ -19,6 +19,9 @@ module "emr-sgs" {
 
 module "emr-iam" {
   source                            = "./modules/aws-emr-iam"
+
+  depends_on = [module.emr-cluster]
```
Comment (along528) [inline main.tf:23]: This appears to have introduced a cyclic dependency. Not sure how to resolve

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

File: modules/aws-emr-iam/main.tf [hunk 2]
```diff
@@ -130,15 +299,145 @@ data "aws_iam_policy_document" "emr_ec2_policy" {
...
+      "elasticmapreduce:DescribeCluster",
+      "elasticmapreduce:DescribeEditor",
+      "elasticmapreduce:DescribeJobFlows",
+      "elasticmapreduce:DescribeNotebookExecution",
+      "elasticmapreduce:DescribeRepository",
+      "elasticmapreduce:DescribeSecurityConfiguration",
+      "elasticmapreduce:DescribeStep",
+      "elasticmapreduce:DescribeStudio",
       "elasticmapreduce:ListBootstrapActions",
       "elasticmapreduce:ListClusters",
```
[the remark this answers is not in the document]
Comment (souza-dan) [reply modules/aws-emr-iam/main.tf:327]: image

[the remark this answers is not in the document]
Comment (souza-dan) [reply modules/aws-emr-iam/main.tf:327]: There's a chance we don't actually need any of these actions at the ec2 level (at least based on the access advisor).

My recommendation is that we just don't create this policy and let the user of the module add any additional policies that may be needed.

Which is kind of the existing model of passing in the s3 policy ARNs.

**Pull request closed without merging**